lib/queue_classic_plus/base.rb

```diff
@@ -165,5 +189,26 @@ def self.deserialized(args)
     def self.execute(sql, *args)
       QC.default_conn_adapter.execute(sql, *args)
     end
+
+    def self.define_callback(wrapped_method, type, callback_method, &_block)
+      callback_module = Module.new
+
+      callback_module.send(:define_method, wrapped_method) do |*args|
+        callback_args = args
+        if wrapped_method == :enqueue
+          callback_args = callback_args.drop(1) # Class/method is the first argument. Callbacks don't need that.
+        end
+
+        if [:before, :around].include?(type)
+          block_given? ? yield(*callback_args) : send(callback_method, *callback_args)
+        end
+        super(*args)
+        if [:after, :around].include?(type)
+          block_given? ? yield(*callback_args) : send(callback_method, *callback_args)
+        end
+      end
+
+      singleton_class.prepend(callback_module)
```

Comment:
I think you could golf this a little:
```ruby
def self.define_callback(wrapped_method, callback_type, callback_method, &_block)
  singleton_class.prepend Module.new do
    define_method wrapped_method do |*args|
      ...
    end
  end
```

Reply:
Nice. It's more expression based than procedural, which is what I had. ✅